### PR TITLE
Implemented a few methods in symengine/eval_arb.cpp

### PR DESCRIPTION
**Pull request by ShikharJ**
This PR is basically a rebase to master + additional minor changes to #861 by @saroad.
As always, please review and suggest improvements.

### CODE REVIEW

**Comment by ShikharJ** [comment]
Ping @isururf.

**Comment by ShikharJ** [comment]
@isuruf How can [this](https://travis-ci.org/ShikharJ/symengine/jobs/219717946) error be fixed?

File: symengine/tests/eval/test_eval_arb.cpp [hunk 1]
```diff
@@ -756,3 +765,186 @@ TEST_CASE("Min/Max: eval_arb", "[eval_arb]")
...
+{
+    arb_t a;
+    arb_init(a);
+
+    RCP<const Basic> r1 = zeta(integer(-34));
+    eval_arb(a, *r1, 13);
+
+    mpfr_t f;
+    mpfr_init2(f, 17);
+    mpfr_set_d(f, eval_double(zeta(integer(-34))), MPFR_RNDN);
```
Comment (isuruf) [inline symengine/tests/eval/test_eval_arb.cpp]: `eval_double(*zeta(integer(-34)))`

File: symengine/eval_arb.cpp [hunk 2]
```diff
@@ -136,7 +136,7 @@ class EvalArbVisitor : public BaseVisitor<EvalArbVisitor>
         throw NotImplementedError("Not Implemented");
     }
 
-    void bvisit(const Complex &)
+    void bvisit(const Complex &x)
```
Comment (isuruf) [inline symengine/eval_arb.cpp]: Remove the redundant `x` here and below

**Comment by ShikharJ** [comment]
Updated @isuruf. Please take a look.

**Comment by ShikharJ** [comment]
Ping @isuruf.

**Comment by ShikharJ** [comment]
Ping  @isuruf.

**Comment by ShikharJ** [comment]
Thanks for the review and merge @isuruf @ranjithkumar007.

**Pull request merged**